src/module/cache-container/counters/CountersCtrl.ts

```diff
@@ -0,0 +1,44 @@
+import {ICacheContainer} from "../../../services/container/ICacheContainer";
+import IModalService = angular.ui.bootstrap.IModalService;
+import {ICounter} from "../../../services/counters/ICounter";
+import {CounterCreateModalCtrl} from "./CounterCreateModalCtrl";
+import {CounterService} from "../../../services/counters/CounterService";
+import {openConfirmationModal} from "../../../common/dialogs/Modals";
+import {IStateService} from "angular-ui-router";
+
+export class CountersCtrl {
+  static $inject: string[] = ["$state", "$uibModal", "container", "counterService", "counters"];
+
+  constructor(private $state: IStateService,
+              private $uibModal: IModalService,
+              private container: ICacheContainer,
+              private counterService: CounterService,
+              private counters: ICounter []) {
+
+  }
+
+  createCounterModal(): void {
+    this.$uibModal.open({
+      templateUrl: "module/cache-container/counters/view/counters-create.html",
+      controller: CounterCreateModalCtrl,
+      controllerAs: "modalCtrl",
+      resolve: {
+        container: (): ICacheContainer => this.container,
+        counterService: (): CounterService => this.counterService
+      }
+    });
+  }
+
+  reset(counter: ICounter): void {
+    this.counterService.reset(this.container, counter).then(() => this.$state.reload());
```

Comment:
This should probably be in a confirmation modal

Reply:
I thought about it as well but then I remembered Heather telling me how only resource create/destruct and operations that leave the system in a different state should be confirmed to avoid modal overclicking. It is fine art to decide this one...

Reply:
Surely resetting a counter will leave the system in a different state?

Reply:
But then what do we do about cache statistic resets that do not have modal. Well, not the same thing...ok I see. You are right!